subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/LauncherServices.java

```diff
@@ -89,6 +91,10 @@ ExecuteBuildActionRunner createExecuteBuildActionRunner() {
         ClassLoaderCache createClassLoaderCache() {
             return new ClassLoaderCache();
         }
+
+        ScheduledExecutorService createScheduledExecutorService() {
+            return Executors.newScheduledThreadPool(1);
```

Comment:
Do we intend this to be reused by other services? Would a single thread still be enough in that case?

If not, instantiating a timer just for the memory info broadcaster might be simpler.